src/schema/term.rs

```diff
@@ -170,6 +162,10 @@ impl Term {
         self.set_bytes(val.to_u64().to_be_bytes().as_ref());
     }
 
+    /// Append a type marker + fast value to a term.
+    /// This is used in JSON type to append a fast value after the path.
+    ///
+    /// It will not clear existing bytes.
     pub(crate) fn append_type_and_fast_value<T: FastValue>(&mut self, val: T) {
         self.0.push(T::to_type().to_code());
```

Comment:
can we have debug assert here too?

Reply:
In indexing we don't have a path, but a path id, so the debug_assert doesn't work here. It's also atypical for the Term methods, they are just building blocks, and the caller has to know the Term state

Reply:
the comment made me think it was only for json. So it is not?

Reply:
It is, but there are two JSON currently 
Indexing: [FieldId][PathId][Value]
Query: [FieldId][JsonPath][JsonPathEnd][Value]

When we switch to `[PathId][Value]` for all data during Indexing, we can split `Term` into `Term` and `IndexingTerm` with a clearer API.